tools/spectral/ipa/rulesets/functions/eachCustomMethodMustBeGetOrPost.js

```diff
@@ -0,0 +1,29 @@
+import { isCustomMethod } from './utils/resourceEvaluation.js';
+
+const ERROR_MESSAGE = 'The HTTP method for custom methods must be GET or POST.';
+const ERROR_RESULT = [{ message: ERROR_MESSAGE }];
+const VALID_METHODS = ['get', 'post'];
+const EXCEPTION_EXTENSION_KEY = 'x-xgen-IPA-exception';
+
+export default (input, opts, { path }) => {
+  // Extract the path key (e.g., '/a/{exampleId}:method') from the JSONPath.
+  let pathKey = path[1];
+
+  if (!isCustomMethod(pathKey)) return;
+
+  //Exclude exception extension key
+  let keys = Object.keys(input);
+  const httpMethods = keys.filter((key) => key !== EXCEPTION_EXTENSION_KEY);
```

Comment:
@lovisaberggren I excluded the exception extension key here, but in the opposite I can only check the list of predefined HTTP methods as well

Reply:
Thanks! Yeah perhaps the opposite is more stable, as there may be other extensions for example added to paths, and HTTP methods probably won't change a lot 😄